`cli-altsrc` and `cli-docs` version drift and incompatibilities

## What's the problem?

The latest `urfave/cli` (v3.0.0-alpha9.2), is not compatible with the latest `urfave/cli-docs` (v3.0.0-alpha5).

Attempting to build a project with those 2 dependencies results in:

```
~/projects/emu git:[master]
go build ./cmd/emu/
# github.com/urfave/cli-docs/v3
../../.cache/go/pkg/mod/github.com/urfave/cli-docs/v3@v3.0.0-alpha5/docs.go:273:16: flag.GetValue undefined (type cli.DocGenerationFlag has no field or method GetValue)
../../.cache/go/pkg/mod/github.com/urfave/cli-docs/v3@v3.0.0-alpha5/docs.go:400:21: flag.GetValue undefined (type cli.DocGenerationFlag has no field or method GetValue)
```

for a sample reproducer, see https://github.com/bartekpacia/emu/pull/7

## What's the larger problem?

There's no CI checks that prevent this kind of problem.

Ideally, nothing that breaks compatibility between the `urfave/cli` modules (`cli`, `altsrc`, and `docs`) should be mergeable (i.e. CI should be red), unless explicit approval from maintainers.

--- a/flag.go
+++ b/flag.go
@@ -129,6 +129,10 @@ type DocGenerationFlag interface {
 	// GetUsage returns the usage string for the flag
 	GetUsage() string
 
+	// GetValue returns the flags value as string representation and an empty
+	// string if the flag takes no value at all.
+	GetValue() string
+
 	// GetDefaultText returns the default text for this flag
 	GetDefaultText() string
 

--- a/flag_impl.go
+++ b/flag_impl.go
@@ -92,7 +92,7 @@ type FlagBase[T any, C any, VC ValueCreator[T, C]] struct {
 // GetValue returns the flags value as string representation and an empty
 // string if the flag takes no value at all.
 func (f *FlagBase[T, C, V]) GetValue() string {
-	if reflect.TypeOf(f.Value).Kind() == reflect.Bool {
+	if !f.TakesValue() {
 		return ""
 	}
 	return fmt.Sprintf("%v", f.Value)

--- a/flag_impl.go
+++ b/flag_impl.go
@@ -89,6 +89,15 @@ type FlagBase[T any, C any, VC ValueCreator[T, C]] struct {
 	value      Value // value representing this flag's value
 }
 
+// GetValue returns the flags value as string representation and an empty
+// string if the flag takes no value at all.
+func (f *FlagBase[T, C, V]) GetValue() string {
+	if reflect.TypeOf(f.Value).Kind() == reflect.Bool {
+		return ""
+	}
+	return fmt.Sprintf("%v", f.Value)
+}
+
 // Apply populates the flag given the flag set and environment
 func (f *FlagBase[T, C, V]) Apply(set *flag.FlagSet) error {
 	tracef("apply (flag=%[1]q)", f.Name)

--- a/flag_test.go
+++ b/flag_test.go
@@ -3127,3 +3127,9 @@ func TestEnvHintWindows(t *testing.T) {
 		assert.Equal(t, "something [%foo%, %bar%, %ss%]", withEnvHint([]string{"foo", "bar", "ss"}, "something"))
 	}
 }
+
+func TestDocGetValue(t *testing.T) {
+	assert.Equal(t, "", (&BoolFlag{Name: "foo", Value: true}).GetValue())
+	assert.Equal(t, "", (&BoolFlag{Name: "foo", Value: false}).GetValue())
+	assert.Equal(t, "bar", (&StringFlag{Name: "foo", Value: "bar"}).GetValue())
+}

--- a/godoc-current.txt
+++ b/godoc-current.txt
@@ -560,6 +560,10 @@ type DocGenerationFlag interface {
 	// GetUsage returns the usage string for the flag
 	GetUsage() string
 
+	// GetValue returns the flags value as string representation and an empty
+	// string if the flag takes no value at all.
+	GetValue() string
+
 	// GetDefaultText returns the default text for this flag
 	GetDefaultText() string
 
@@ -694,6 +698,10 @@ func (f *FlagBase[T, C, V]) GetEnvVars() []string
 
 func (f *FlagBase[T, C, V]) GetUsage() string
     GetUsage returns the usage string for the flag
+
+func (f *FlagBase[T, C, V]) GetValue() string
+    GetValue returns the flags value as string representation and an empty
+    string if the flag takes no value at all.
 
 func (f *FlagBase[T, C, V]) IsDefaultVisible() bool
     IsDefaultVisible returns true if the flag is not hidden, otherwise false

--- a/testdata/godoc-v3.x.txt
+++ b/testdata/godoc-v3.x.txt
@@ -560,6 +560,10 @@ type DocGenerationFlag interface {
 	// GetUsage returns the usage string for the flag
 	GetUsage() string
 
+	// GetValue returns the flags value as string representation and an empty
+	// string if the flag takes no value at all.
+	GetValue() string
+
 	// GetDefaultText returns the default text for this flag
 	GetDefaultText() string
 
@@ -694,6 +698,10 @@ func (f *FlagBase[T, C, V]) GetEnvVars() []string
 
 func (f *FlagBase[T, C, V]) GetUsage() string
     GetUsage returns the usage string for the flag
+
+func (f *FlagBase[T, C, V]) GetValue() string
+    GetValue returns the flags value as string representation and an empty
+    string if the flag takes no value at all.
 
 func (f *FlagBase[T, C, V]) IsDefaultVisible() bool
     IsDefaultVisible returns true if the flag is not hidden, otherwise false

